src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -2017,6 +2017,47 @@ string[] linkingTargetFields
             }
         }
 
+        /// <summary>
+        /// This test checks that the final config used by runtime engine doesn't lose the directory information
+        /// if provided in the base config file path when loading the config based on the environment.
+        /// </summary>
+        /// <param name="addDirectory"></param>
+        /// <param name="environmentValue"></param>
+        /// <param name="baseConfigFilePath"></param>
+        /// <param name="finalConfigFilePath"></param>
+        [DataTestMethod]
+        [DataRow(false, "", false, "current-dir-dab-config.json", "current-dir-dab-config.json", DisplayName = "Config file in the current directory")]
+        [DataRow(true, "", false, "test\\diff-dir-dab-config.json", "test\\diff-dir-dab-config.json", DisplayName = "Config file in a different directory")]
+        [DataRow(false, "Test", false, "current-dir-dab-config.json", "current-dir-dab-config.Test.json", DisplayName = "Config file in the current directory with non-empty environment value")]
+        [DataRow(true, "Test", false, "test\\diff-dir-dab-config.json", "test\\diff-dir-dab-config.Test.json", DisplayName = "Config file in a different directory with non-empty environment value")]
+        [DataRow(false, "", true, "current-dir-dab-config.json", "current-dir-dab-config.json", DisplayName = "Config file in the current directory using absolute path")]
+        [DataRow(true, "", true, "test\\diff-dir-dab-config.json", "test\\diff-dir-dab-config.json", DisplayName = "Config file in a different directory using absolute path")]
+        [DataRow(false, "Test", true, "current-dir-dab-config.json", "current-dir-dab-config.Test.json", DisplayName = "Config file in the current directory with non-empty environment value using absolute path")]
+        [DataRow(true, "Test", true, "test\\diff-dir-dab-config.json", "test\\diff-dir-dab-config.Test.json", DisplayName = "Config file in a different directory with non-empty environment value using absolute path")]
+        public void TestDirectoryInfoIsRetainedInFinalConfig(
+            bool IsDifferentDirectory,
+            string environmentValue,
+            bool isAbsolutePath,
+            string baseConfigFilePath,
+            string finalConfigFilePath)
+        {
+            MockFileSystem fileSystem = new();
+            if (IsDifferentDirectory)
+            {
+                fileSystem.AddDirectory("test");
```

Comment:
In `runtimeConfigLoader.GetFileName(...)` there is a call to `_fileSystem.Directory.GetCurrentDirectory();`. Is the current directory in the mock file system representative of the value returned in a real Az Container Apps deployment? 

The PR description is not clear about what change after `0.7.6` caused the bug this PR is fixing. The requirement to now supply a 'full' file path looks to make everything work, but it is not clear why it works because per the container deploy script, the config file is uploaded to the root of the config file share:
https://github.com/Azure/data-api-builder/blob/f8373b8e231ddae1f8bba99160f6de1932d9f4a6/samples/azure/azure-container-apps-deploy.sh#L50-L52

What would be an example of the full file path provided in Az container apps? do you need to know the mount name/location of the file share that is created?

Reply:
> What would be an example of the full file path provided in Az container apps? do you need to know the mount name/location of the file share that is created?

As I understand how the `az storage file upload` command works when uploading to an azure file share, the full path would be made up of <mount-location>/<file-share-name>/<directory-path>/<filename>

so if the mount location was `/mnt/azfiles`, and the file share name was `fileshare`, and the directory path was `DAb/config`, and the file name was `config.json`, then the full path would be

`/mnt/azfiles/fileshare/DAb/config/config.json`

I suppose this is something we can document, but I guess that the person using the az file share should know what a full path is within this context.

Reply:
@Sean, when deploying in Azure Container Apps, we need to mount the volume as well from Azure Container Environment, so if the mounted volume contained the config in `/mnt/configs.dab-config.json`, then we need to provide the correct path in the `--ConfigFileName`. Due to changes in 0.8 release, the directory info was getting lost and we were left with `dab-config.json`, and hence getting config file not found error.

Reply:
updated PR description

Reply:
Okay, I was confused because of the usage of the term "file name" which to me would be `dab-config.json`. If there are path components included like `mnt/folder/dab-config.json`, i would  use the term 'full path' or 'relative path' depending on the path..